### ISSUE TITLE
Unbalanced eval brackets causes TF parser loop

### ISSUE THREAD

**Issue by robeden** [bridgecrewio/checkov#822]
**Describe the bug**
From this ([link 1](https://codifiedsecurity.slack.com/archives/CSMKB8XPH/p1611741949003000), [link 2](https://codifiedsecurity.slack.com/archives/CSMKB8XPH/p1611743126004100)) slack conversation, a malformed eval statement with mismatched brackets can cause a loop in the TF parser. Example statement (note missing closing bracket):
```
s3_access_logs_prefix = "${replace(var.cdn_logging_prefix, "cdn", "s3")/${var.bucket_name}"
```

It is currently unknown if it is an infinite loop, but it's at least very slow.

**To Reproduce**
I have reproduced with this parse scenario:
_main.tf_:
```
locals {
  # This is intentionally missing the closing bracket
  s3_access_logs_prefix = "${replace(var.cdn_logging_prefix, "cdn", "s3")/${var.bucket_name}"
}
```

**Expected behavior**
_expected.json_:
```
{
  "main.tf": {
    "locals": [
      {
        "s3_access_logs_prefix": ["${replace(var.cdn_logging_prefix, 'cdn', 's3')/${var.bucket_name}"]
      }
    ]
  }
}
```
I wouldn't expect extra looping in the parser.

**Comment by robeden** [comment]
I'm looking into this...

**Comment by robeden** [comment]
@schosterbarak - Looks like the loop is in the hcl2 parser. It never returns from the `hcl2.load(f)` call.

**Comment by robeden** [comment]
I've put a test case here ☝️ . The problem is in the `python-hcl2` library (or at least the BC fork of it), so we'll need some fix there.

**Comment by metahertz** [comment]
🤘 Rock on! Thanks Rob, know you've been speaking with Barak+Ariel on the issue.

Adding a bit of context for other watchers on the issue:
Parser in use by checkov is here: https://github.com/bridgecrewio/python-hcl2 Assigning @arielkru and work-in-progress

**Comment by arielkru** [comment]
Hi @robeden & @metahertz !
I've been looking for several ways to solve this issue, but didn't find a way that wouldn't break other well-written tf files, and since this issue is based on a broken tf format anyway, I think we shouldn't change the parser, but would like to know if you have other ideas, here are the details:
* Currently the parser identifies anything that starts with `"` as a string, unless it starts with `"&{` and then it's parsed as an interpolation. The way the parser knows the interpolation ends and the next token starts is by finding the closing `}`.
* I thought of two ways to parse the behaviour of interpolations that start with `"&{` but don't end with `}` like in the example above:
1. **Extending the Interpolations format**: If a string starts with `"${ `it can end properly with `}`, or by the end of the line (`\n`). This solves the currsnt issue, but would break any interpolation that is written as a string that is divided to several lines. 
2. **Parsing it as a string**: the other option is just defining it as a string and not as an interpoaltion, but then it would break on the first next `"` the parser finds, because that's the sign for closing the string on the left of it and parsing a new token, but the thing about interpolations (like the one in the example) is that they often contain sub-strings inside them, and that's why the parser is currently looking for the `}` to know an interpolation ends.

Bottom line is I'm currently thinging we better leave the behaviour as it is. It's not a common issue AFAIK, and the tf file in the current issue is anyway broken, so it makes sense to me that in order to keep the parser aligned to hcl's logic, we shouldn't handle it.
LMK what you think!

**Comment by robeden** [comment]
@arielkru - Are you able to find where the infinite loop happens and maybe error out (don't know if it's a `lark` thing or a `python-hcl2` thing)? I don't think we need to worry about parsing it, but it would be great if it didn't hang forever.

**Comment by arielkru** [comment]
@robeden Good idea, I'm not sure about it but I'll check that out

**Comment by metahertz** [comment]
To @robeden's point, at the very least it would be good if `LOG_LEVEL=DEBUG` had outputted the file path before hanging so we have some pointer to the issue in the users logs.

**Comment by Saarett** [comment]
Hi everyone,
We've added a timeout for the HCL2 parsing method by using a child process - it has 60 seconds to parse the file or otherwise the process is terminated. Unparsed files will be added to the report as:
`Error parsing file <file_path>`
Feel free to share your thoughts on this

**Comment by gruebel** [comment]
I tested it and it doesn't hang anymore, therefore I'm closing the issue, but feel free to open a new one, if there is an other problem with a hanging scan.
